src/_guides/libraries/java-interop.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Java interop using package:jnigen"
+description: "To use Java in your Dart program, use package:jnigen."
+jnigen: "https://pub.dev/packages/jnigen"
+jni: "https://pub.dev/packages/jni"
+example: "https://github.com/HosseinYousefi/jnigen_example/tree/main"
+jnidoc: "https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/jniTOC.html"
+---
+
+Dart mobile, command-line, and server apps
+running on the [Dart Native platform](/overview#platform), on Android, Windows
+and Linux can use [`package:jni`]({{page.jni}})
+and [`package:jnigen`]({{page.jnigen}})
+to call Java and Kotlin APIs.
+
+{{site.alert.note}}
+  This interop feature is **experimental**,
+  and [in active development](https://github.com/dart-lang/jnigen/issues).
+{{site.alert.end}}
+
+`package:jni` allows Dart code to interact
+with Java through [JNI]({{page.jnidoc}}). 
+However, doing so involves a lot of boilerplate code,
+so you can use `package:jnigen` to automatically generate
+the Dart bindings for a given Java API.
+
+You can decompile Kotlin to Java, allowing `package:jnigen`
+to generate bindings for Kotlin as well.
+
+## Simple Java Example
+
+This guide walks you through [an example]({{page.example}})
+that uses `package:jnigen` to generate bindings for a simple class
+that can show a toast.
+
+Along with JDK, maven is required. 
+
+It's recommended to have `clang-format` installed to format the generated
+C bindings.
```

Comment:
Consider putting this into a tip:

```suggestion
{{site.alert.ti}}
  It's recommended to have `clang-format` installed to
  format the generated C bindings.
{{site.alert.end}}
```

Also consider cross-linking to more information about `clang-format` as well.

Reply:
I think the best way to do this would be a "Prerequisites" section, for this line, and the JDK/Maven line. And link to each

> ### Simple Java Example
> This guide walks you through [an example](https://github.com/dart-lang/site-www/blob/main/src/_guides/libraries/%7B%7Bpage.example%7D%7D) that uses `package:jnigen` to generate bindings for a simple class that can show a toast.
> 
> #### Prerequisites
> - [JDK]()
> - [Maven]()
> - (Optional) [`clang-format`]() to format the generated C bindings
> 
> ### Configuring jnigen

Reply:
Love it!